include/seccomp.h.in

```diff
@@ -198,44 +198,108 @@ struct scmp_arg_cmp {
  */
 #define SCMP_SYS(x)		(__NR_##x)
 
+/* Helper macros for the 32-bit argument comparison macros */
+/* Expand to the number of arguments given */
+#define SCMP_VA_NUM_ARGS(...)	SCMP_VA_NUM_ARGS_IMPL(__VA_ARGS__,2,1)
+#define SCMP_VA_NUM_ARGS_IMPL(_1,_2,N,...)	N
+
+/* Expand to alternative names based on the number of arguments given. */
+#define SCMP_MACRO_DISPATCHER(func, ...) \
+	SCMP_MACRO_DISPATCHER_IMPL1(func, SCMP_VA_NUM_ARGS(__VA_ARGS__))
+#define SCMP_MACRO_DISPATCHER_IMPL1(func, nargs) \
+	SCMP_MACRO_DISPATCHER_IMPL2(func, nargs)
+#define SCMP_MACRO_DISPATCHER_IMPL2(func, nargs) \
+	func ## nargs
+
+/* Implement casting of operands to uint32_t */
+#define SCMP_CMP32_1(x, y, z)		SCMP_CMP64(x, y, (uint32_t)(z))
+#define SCMP_CMP32_2(x, y, z, q)	SCMP_CMP64(x, y, (uint32_t)(z), (uint32_t)(q))
```

Comment:
These two macros are borderline, but I'm leaning towards treating them like the others: drop the comment, add the leading underscore, drop the vertical whitespace.